### ISSUE TITLE
A Modular Monolith in Laravel Lumen

### ISSUE THREAD

**Issue by simonhamp** [laravel/blog-contest-may-mayhem#6]
https://medium.com/@simonhamp/a-modular-monolith-in-laravel-lumen-4933b8cab2f1

**Comment by shane-smith** [comment]
Was an enjoyable read.

**Comment by simonhamp** [comment]
Thanks @shane-smith 🙂

**Comment by erikmellum** [comment]
Yup definitely cool. I also totally agree. I think this is a wonderful way to build an app.

**Comment by simonhamp** [comment]
@erikmellum Yep it's pretty great!

**Comment by rasmuscnielsen** [comment]
Awesome post man, great read!

I started structuring my projects in a similar way about a year ago, and it totally reframed my mindset   in terms of decoupling. Can't recommend it enough.

Would love to hear more about your folder structure, migrations, tests, git, etc.! 

Btw 👍 for adopting the 'support' naming from Laravel, much better than the 'common' I came up with ;-)

**Comment by simonhamp** [comment]
Thanks @rasmuscnielsen! Thanks for reading.

Yeh it’s a cool approach. Has been super beneficial for us. It has its place... I think it can be a lot of trouble to go to for smaller projects and there are a few hurdles when you’ve got packages interconnected that need to talk to each other...

I’ll try to write up a post on all that other stuff at some point.
